### DISCUSSION TITLE
importActual fails for a typescript workspace

### DISCUSSION THREAD

**Discussion by behnoodk** [vitest-dev/vitest#1722] [Q&A] [answered]
Hi. I am encountering a weird problem and I'm quite sure I'm doing something wrong that's why I did not create an issue. if it turns out to be a bug, I'll create an issue.

So if I call `vi.importActual` for a yarn workspace that has  the main script outside of src folder, it fails to resolve the file. I have created a minimal repo to demonstrate this.

https://github.com/behnoodk/vitest-importactual

steps to reproduce:
1. `yarn install`
2. `yarn workspace @myorg/testsrv test`

imortActual works as expected for the package `@myorg/testpkg` but fails for `@myorg/testpkgts`. the error message printed is this:

```
 FAIL  index.test.ts [ index.test.ts ]
Error: Error: Cannot find module /Users/mohsen/Documents/projects/vitest-importactual/services/testsrv/Users/mohsen/Documents/projects/vitest-importactual/packages/testpkgts/dist/index.js imported from file:///Users/mohsen/Documents/projects/vitest-importactual/services/testsrv, file:///Users/mohsen/Documents/projects/vitest-importactual/services/testsrv/node_modules
```

So it's looking for the entrypoint of package in the wrong place.

I'd appreciate if someone more familiar with vitest could point me in the right direction.

**Comment by sheremet-va** [comment, marked as answer]
Looks like a bug to me 🤔 It has root two times 🤔

**Reply by behnoodk** [reply]
Thank you for your reply. I'll convert this into an issue so the problem can be looked further into.
